src/compiler/checker.ts

```diff
@@ -34019,7 +34019,16 @@ namespace ts {
 
         function isFunctionUsedInBinaryExpressionChain(node: Node, testedSymbol: Symbol): boolean {
             while (isBinaryExpression(node) && node.operatorToken.kind === SyntaxKind.AmpersandAmpersandToken) {
-                if (isCallExpression(node.right) && testedSymbol === getSymbolAtLocation(node.right.expression)) {
+                const isUsed = forEachChild(node.right, function visit(child): boolean | undefined {
+                    if (isIdentifier(child)) {
+                        const symbol = getSymbolAtLocation(child);
+                        if (symbol && symbol === testedSymbol) {
+                            return true;

```

Comment:
Why don't we need the additional checks from isFunctionUsedInConditionBody here? I think the range of things that can appear on the lhs of a `&&` is the same as can appear in the condition of an `if`.

Reply:
Not sure if we need to add additional checks that `isFunctionUsedInConditionBody` uses to test the same expression target, because, in a `BinaryExpression` context,  we just need to need to find the same symbol in all expressions from the right side ` && `. Maybe you have additional test cases that need to be covered?

Reply:
@sandersn Not sure if we need to add additional checks that `isFunctionUsedInConditionBody` uses to test the same expression target, because, in a `BinaryExpression` context,  we just need to need to find the same symbol in all expressions from the right side ` && `, or am I wrong?.  Maybe you have additional test cases that need to cover?

Reply:
From the equivalent tests in truthinessCallExpressionCoercion*:
`x.foo.bar && x.foo.bar()`
`x.foo.bar && x.foo.bar++`
`this.bunch && this.bunch()`

Reply:
But the new test already has those cases, and they pass, so this must be good enough!

Reply:
@sandersn Thanks for the feedback. I think such tests exist

1. https://github.com/microsoft/TypeScript/blob/335b11cd5163201afd4171aad95ef089c386f1e6/tests/cases/compiler/truthinessCallExpressionCoercion2.ts#L66-L67
2. `x.foo.bar && x.foo.bar++` - it seems `bar` is not a `function`
3. https://github.com/microsoft/TypeScript/blob/335b11cd5163201afd4171aad95ef089c386f1e6/tests/cases/compiler/truthinessCallExpressionCoercion2.ts#L99-L100

It is ok?

Reply:
@sandersn I didn't notice your previous comment., you are fast :).

Reply:
@sandersn Can I mark this comment as resolved?